test/common.js

```diff
@@ -12,6 +12,61 @@ exports.tmpDirName = 'tmp';
 exports.PORT = +process.env.NODE_COMMON_PORT || 12346;
 exports.isWindows = process.platform === 'win32';
 
+var rimrafSync = function(p) {
+  try {
+    var st = fs.lstatSync(p);
+  } catch (e) {
+    if (e.code === 'ENOENT')
+      return;
+  }
+
+  try {
+    if (st && st.isDirectory())
+      rmdirSync(p, null);
+    else
+      fs.unlinkSync(p);
+  } catch (e) {
+    if (e.code === 'ENOENT')
+      return;
+    if (e.code === 'EPERM')
+      return rmdirSync(p, er);
+    if (e.code !== 'EISDIR')
+      throw e;
+    rmdirSync(p, e);
+  }
+};
+
+var rmdirSync = function(p, originalEr) {
+  try {
+    fs.rmdirSync(p);
+  } catch (e) {
+    if (e.code === 'ENOENT')
+      return;
+    if (e.code === 'ENOTDIR')
+      throw originalEr;
+    if (e.code === 'ENOTEMPTY' || e.code === 'EEXIST' || e.code === 'EPERM') {
+      fs.readdirSync(p).forEach(function(f) {
+        rimrafSync(path.join(p, f));
+      });
+      fs.rmdirSync(p);
+    }
+  }
+};
+
+var refreshTmpDir = function() {
+  if (!process.send) { // Not a child process
+    try {
+      rimrafSync(exports.tmpDir);
+    } catch (e) {
+    }
+
+    try {
+      fs.mkdirSync(exports.tmpDir);
```

Comment:
I agree, but I was trying to keep this change as small as possible. The current Python script swallows errors if `mkdir` fails, so I did the same. (Actually, it swallows errors and keeps retrying if the directory doesn't exist. Comment says that's because Python `mkdir` on win32 fails intermittently. I was going to wait and see if that actually crops up as an issue with `fs.mkdirSync` before duplicating that functionality.)